app/reducers/gallery.js

```diff
@@ -24,12 +29,42 @@ export default function gallery(state=initialState, action) {
       return {
         status: Status.PENDING,
         photos: state.photos,
+        collection: state.collection,
         error: null
       };
     case DID_FINISH_FETCHING_PHOTOS:
       return {
         status: Status.SUCCESS,
         photos: shuffle(state.photos.concat(action.photos)),
+        collection: state.collection,
+        error: null
+      };
+    case FIREBASE_DID_UPDATE:
+      let permanentsObj = action.snapshot.val().permanents;
+      let permanents = [];
+      for(let firebaseId in permanentsObj) { 
+        if (permanentsObj.hasOwnProperty(firebaseId)) {
+          let mediaObj = permanentsObj[firebaseId];
+          mediaObj.id = firebaseId;
+          permanents.push(mediaObj);
+        }
+      }
+
+      let orderedPermanents = permanents.sort(function (a, b) {
+        if (a.bumpCount > b.bumpCount) {
+          return -1;
+        }
+        if (a.bumpCount < b.bumpCount) {
+          return 1;
+        }
+        return 0;
+      });
+      console.log(orderedPermanents)
```

Comment:
I'd prolly write the sort like this:

```
let orderedPermanents = permanents.sort((a, b) => {
  if (a.bumpCount > b.bumpCount) { return -1; }
  if (a.bumpCount < b.bumpCount) { return 1; }
  return 0;
});
```

mainly just wanna b sure we es6-ify everything!  online-liner stuff is optional i dont mind